test/unit/connection_test.rb

```diff
@@ -0,0 +1,96 @@
+require File.join(File.dirname(__FILE__), '../test_helper')
+require 'tmpdir'
+require 'tempfile'
+
+describe HammerCLIForeman::Api do
+  describe '#foreman_api_reconnect' do
+    let(:basic_auth) { HammerCLIForeman::AUTH_TYPES[:basic_auth] }
+    let(:subject) { HammerCLIForeman.foreman_api_reconnect(basic_auth) }
+
+    before do
+      HammerCLI::Settings.load({ :foreman => { :host => "https://foreman.example.com" } })
+    end
+
+    it 'drops any existing connection to foreman' do
+      HammerCLI::Connection.any_instance.expects(:drop).with(HammerCLIForeman::CONNECTION_NAME)
+      subject
+    end
+
+    it 'creates a new connection to foreman' do
+      HammerCLI::Connection.any_instance.expects(:create).with(HammerCLIForeman::CONNECTION_NAME)
+      subject
+    end
+  end
+end
```

Comment:
@ofedoren can you please take a look at these test cases? They seem to pass in the file but all the other files test cases are failing due to this change. I think it is mostly because of the the `:drop` method. Once dropped it is unable to create a connection again.

Reply:
Like `before` block I tried to use the `after` block too to create a connection again but it does not seem to be working that way :(

Reply:
Well.. I think i found why is that:

`HammerCLIForeman.foreman_api_reconnect` creates an actual connection while in tests we create a fake connection:

https://github.com/theforeman/hammer-cli-foreman/blob/master/test/test_helper.rb#L23-L25
and
https://github.com/theforeman/hammer-cli-foreman/blob/637846b3d856b9683657b780c945ae678085b027/lib/hammer_cli_foreman/testing/api_expectations.rb#L153-L161